diff --git a/dist/index.mjs b/dist/index.mjs
index 42fd3b1fe98c245d5a1491d1f15434b70a06eacc..f1051b7463be32730d8e3cf2dfc1c221724345f5 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -304,7 +304,7 @@ function useChat({
         );
         const existingData = streamDataRef.current;
         await callChatApi({
-          api,
+          api: extraMetadataRef.current.api || api,
           body: (_b = experimental_prepareRequestBody == null ? void 0 : experimental_prepareRequestBody({
             id: chatId,
             messages: chatMessages,
@@ -499,7 +499,7 @@ function useChat({
       triggerRequest(chatRequest);
       setInput("");
     },
-    [input, generateId2, triggerRequest]
+    [input, generateId2, triggerRequest, api]
   );
   const handleInputChange = (e) => {
     setInput(e.target.value);
@@ -583,12 +583,14 @@ function useCompletion({
     extraMetadataRef.current = {
       credentials,
       headers,
-      body
+      body,
+      api
     };
-  }, [credentials, headers, body]);
+  }, [credentials, headers, body, api]);
   const triggerRequest = useCallback3(
-    async (prompt, options) => callCompletionApi({
-      api,
+    async (prompt, options) => { 
+      return callCompletionApi({
+      api: extraMetadataRef.current.api || api,
       prompt,
       credentials: extraMetadataRef.current.credentials,
       headers: { ...extraMetadataRef.current.headers, ...options == null ? void 0 : options.headers },
@@ -613,7 +615,8 @@ function useCompletion({
       onResponse,
       onFinish,
       onError
-    }),
+    });
+    },
     [
       mutate,
       mutateLoading,
